tests/lib/cpp/cxx/src/main.cpp

```diff
@@ -106,6 +106,7 @@ BUILD_ASSERT(ARRAY_SIZE(foos) == 5, "expected 5 elements");
 /* Check that SYS_INIT() compiles. */
 static int test_init(void)
 {
+	(void)foos;
```

Comment:
why do we need a random cast here? this doesn't look to be doing anything at all?

Reply:
isn't this how one ignores return values? odd though, it's marked as __maybe_unused, I can try to drop it, again I pulled it from the other PR, can't recall if this was erroring out or not

Reply:
Sure but this isn't calling a function is it? its casting a value to (void) which would be a do nothing operation I'd think? This is really confusing to read.

Reply:
let me try and drop it, may not be needed after all for whatever reason

Reply:
alright gone, let's see if CI is still happy, can't repro locally so it may be alright